src/index.ts

```diff
@@ -44,23 +46,116 @@ export default class WorkspaceClient {
         return new RemoteAPI(resources);
     }
 
-    public static getJsonRpcApi(entryPoint: string): IWorkspaceMasterApi {
-        const transport = new WebSocketClient();
-        return new WorkspaceMasterApi(transport, entryPoint);
-    }
-
     private static createAxiosInstance(config: IRestAPIConfig): AxiosInstance {
-        if (config.ssCrtPath && this.isItNode() && fs.existsSync(config.ssCrtPath)) {
-            const agent = new https.Agent({
-                ca: fs.readFileSync(config.ssCrtPath)
-            });
-            return axios.create({httpsAgent: agent});
+        if (!this.isItNode()) {
+            this.addLogInterceptorsIfEnabled(axios, config);
+            return axios;
         }
 
+        const certificateAuthority = this.getCertificateAuthority(config);
+        const proxyUrl = process.env.http_proxy;
+        if (proxyUrl && proxyUrl !== '' && config.baseUrl) {
+            const parsedBaseUrl = url.parse(config.baseUrl);
+            if (parsedBaseUrl.hostname && this.shouldProxy(parsedBaseUrl.hostname)) {
+                const axiosRequestConfig: AxiosRequestConfig | undefined = {
+                    proxy: false,
+                };
+                const parsedProxyUrl = url.parse(proxyUrl);
+                const mainProxyOptions = this.getMainProxyOptions(parsedProxyUrl);
+                const httpsProxyOptions = this.getHttpsProxyOptions(mainProxyOptions, parsedBaseUrl.hostname, certificateAuthority);
+                const httpOverHttpAgent = tunnel.httpOverHttp({ proxy: mainProxyOptions });
+                const httpOverHttpsAgent = tunnel.httpOverHttps({ proxy: httpsProxyOptions });
+                const httpsOverHttpAgent = tunnel.httpsOverHttp({
+                    proxy: mainProxyOptions,
+                    ca: certificateAuthority ? [certificateAuthority] : undefined
+                });
+                const httpsOverHttpsAgent = tunnel.httpsOverHttps(httpsProxyOptions);
```

Comment:
I fear that here you will have to also add the `ca` at the top level, besides the `proxy`argument.
(that's a case we cannot test for now, but that's how tunnel docs shows in the examples).

`ca` at the top-level, when target URL is `https`
`ca` in the proxy settings, when the proxy itself is `https`

Reply:
fixed